### PR TITLE
Add qa tier

### PR DESCRIPTION
**Pull request by ktp242**
This PR sets the configurations for adding the QA tier. The changes include adding config/var_qa.env and update the deploy-qa scripts in package.json. Along with the documentations and the version.

Please see the new CheckinRequestService-qa Lambda here
https://console.aws.amazon.com/lambda/home?region=us-east-1#/functions/CheckinRequestService-qa?newFunction=true&tab=graph

and new checkin-requests-qa RDS here
https://console.aws.amazon.com/rds/home?region=us-east-1#dbinstance:id=checkin-requests-qa

They are both under nypl-digital-dev

The PR also follows the same deploy process as checkout-request-service to use travis-deploy.sh to deploy the instance with Travis. In this way, we can still deploy different environments locally without specify AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY.

samples/checkin_requests-development.sql is the file serves as an example for the schema used in the RDS.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
-    "deploy-production": "./node_modules/.bin/node-lambda deploy -e production -f config/var_production.env -o arn:aws:iam::946183545209:role/lambda-full-access -b subnet-5deecd15,subnet-59bcdd03 -g sg-116eeb60 -a $AWS_ACCESS_KEY_ID_PRODUCTION -s $AWS_SECRET_ACCESS_KEY_PRODUCTION"
+    "deploy-development": "./node_modules/.bin/node-lambda deploy -e development -f config/var_development.env -S config/event_sources_development.json -o arn:aws:iam::224280085904:role/lambda_basic_execution --profile nypl-sandbox",
+    "deploy-qa": "./node_modules/.bin/node-lambda deploy -e qa -f config/var_qa.env -S config/event_sources_qa.json -o arn:aws:iam::946183545209:role/lambda-full-access --profile nypl-digital-dev",
+    "deploy-production": "./node_modules/.bin/node-lambda deploy -e production -f config/var_production.env -S config/event_sources_production.json -o arn:aws:iam::946183545209:role/lambda-full-access --profile nypl-digital-dev"
```
Comment (nonword) [inline package.json]: Doesn't this need to be set up in a VPC to talk to the dbs?

Comment (ktp242) [reply package.json]: I have set up VPC, subnet, and security groups here,
for qa
https://console.aws.amazon.com/lambda/home?region=us-east-1#/functions/CheckinRequestService-qa?newFunction=true&tab=graph

But I am a bit confused about the settings. I checked Checkin Service production
https://console.aws.amazon.com/lambda/home?region=us-east-1#/functions/CheckinRequestService-production?tab=graph

It doesn't seem lined up with Checkin RDS production
https://console.aws.amazon.com/rds/home?region=us-east-1#dbinstance:id=checkin-requests-production

Also, we don't have an RDS for development under nypl-sandbox. The same case happens for Checkout Service tiers as well.

I think I will need some more info to set the configurations correct.

**Comment by ktp242** [comment]
I have updated the subnets and security groups to scripts for deployment. Also the settings on our Lambda instances.

**Comment by ktp242** [comment]
This has finally been deployed to QA! It should be all good for testing now. Please test with the URL https://qa-platform.nypl.org/api/v0.1/checkin-requests

I use the the POST body as below, but other valid data should work too
```
{
  "itemBarcode": "33333410671289",
  "owningInstitutionId": "NYPL"
}
```

A successful response could be as below,

```
{
    "data": {
        "id": 18,
        "success": true,
        "processed": true,
        "createdDate": "2018-09-20T12:47:36-04:00",
        "updatedDate": "2018-09-20T12:47:37-04:00",
        "itemBarcode": "33333410671289",
        "owningInstitutionId": "NYPL",
        "cancelRequestId": null,
        "jobId": "525ba3cf2875b0d"
    },
    "count": 1,
    "statusCode": 208,
    "debugInfo": []
}
```

**Pull request merged**